src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -99,25 +101,27 @@ object Task {
     */
   case class ReservationWithVolumes(volumeIds: Iterable[LocalVolumeId])
 
-  case class LocalVolumeId(idString: String) {
-    override def toString: String = s"LocalVolume [$idString]"
+  case class LocalVolume(id: LocalVolumeId, persistentVolume: PersistentVolume)
 
-    def containerPath: String = {
-      //TODO: make the case class hold all different parts and assemble an id string on demand?
-      idString.split("""\.""")(1)
-    }
+  case class LocalVolumeId(appId: PathId, containerPath: String, uuid: String) {
+    import LocalVolumeId._
+    lazy val idString = appId.safePath + appDelimiter + containerPath + volumeDelimiter + uuid
+
+    override def toString: String = s"LocalVolume [$idString]"
   }
 
   object LocalVolumeId {
-    def apply(appId: PathId, path: String): LocalVolumeId = {
-      //FIXME: mock implementation from ME
-      LocalVolumeId(s"${appId.safePath}.$path.random")
-    }
-
+    private val uuidGenerator = Generators.timeBasedGenerator(EthernetAddress.fromInterface())
+    private val appDelimiter = "."
+    private val volumeDelimiter = "."
     private val LocalVolumeEncoderRE = "^([^.]+).([^.]+).([^.]+)$".r
```

Comment:
Since we have delimiter, we should use them in the regexp:
```scala
private val LocalVolumeEncoderRE = s"^([^.]+)[$appDelimiter]([^.]+)[$volumeDelimiter]([^.]+)$".r
```